.github/workflows/androidTestSuite.yml

```diff
@@ -0,0 +1,126 @@
+        with:
+          distribution: 'adopt'
+          java-version: '11'
+
+      - name: Setup Python 2.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 2.7
+
+      - name: Install Pillow and Codecov
+        run: |
+          python -m pip install --upgrade pip
+          pip install --user codecov
+          pip install --user Pillow==6.2.2
+
+      - name: Gradle cache
+        uses: gradle/gradle-build-action@v2
+
+      - name: AVD cache
+        uses: actions/cache@v2
+        id: avd-cache
+        with:
+          path: |
+            ~/.android/avd/*
+            ~/.android/adb*
+          key: avd-${{ matrix.api-level }}
+
+      - name: Create AVD and generate snapshot for caching
+        if: steps.avd-cache.outputs.cache-hit != 'true'
+        uses: reactivecircus/android-emulator-runner@v2
+        with:
+          profile: Nexus 4
+          api-level: ${{ matrix.api-level }}
+          arch: x86
+          target: google_apis
+          disable-animations: false
+          force-avd-creation: false
+          emulator-options: -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          script: echo "Generated AVD snapshot for caching."
+
+      - name: Run tests
+        if: always()
+        uses: ReactiveCircus/android-emulator-runner@v2
+        with:
+          profile: Nexus 4
+          api-level: ${{ matrix.api-level }}
+          arch: x86
+          target: google_apis
+          disable-animations: true
+          force-avd-creation: false
+          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          script: |
+            chmod +x gradlew
+            ./gradlew jacocoTestProdDebugUnitTestReport
+
+      - name: Run E2E
+        if: always()
+        uses: ReactiveCircus/android-emulator-runner@v2
+        with:
+          profile: Nexus 4
+          api-level: ${{ matrix.api-level }}
+          arch: x86
+          target: google_apis
+          disable-animations: true
+          force-avd-creation: false
+          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          script: |
+            ./gradlew verifyProdDebuggableAndroidTestScreenshotTest -PdisablePreDex
+
+      - name: Run Artifacts
+        uses: ReactiveCircus/android-emulator-runner@v2
+        with:
+          profile: Nexus 4
+          api-level: ${{ matrix.api-level }}
+          arch: x86
+          target: google_apis
+          disable-animations: true
+          force-avd-creation: false
+          emulator-options: -no-snapshot-save -no-window -gpu swiftshader_indirect -noaudio -no-boot-anim -camera-back none
+          script: |
+            ./gradlew copyLintBuildArtifacts
+            ./gradlew copyUnitTestBuildArtifacts
+            ./gradlew pullProdDebuggableAndroidTestScreenshots -PdisablePreDex
```

Comment:
better to merges these steps, so no need to boot the emulator more than one time.